chart/templates/redis/redis-service.yaml

```
@@ -35,7 +35,14 @@ metadata:
       {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
+{{- if (or (eq .Values.redis.service.type "ClusterIP") (empty .Values.redis.service.type)) }}
```

you can remove checking if it's empty because you don't allow empty value in values.schema.json
```suggestion
{{- if eq .Values.redis.service.type "ClusterIP" }}
```

---

Done